addons/stock_account/stock.py

```diff
@@ -315,7 +315,7 @@ def _invoice_create_line(self, cr, uid, moves, journal_id, inv_type='out_invoice
             origin = move.picking_id.name
             partner, user_id, currency_id = move_obj._get_master_data(cr, uid, move, company, context=context)
 
-            key = (partner, currency_id, company.id)
+            key = (partner, currency_id, company.id, user_id)
             invoice_vals = self._get_invoice_vals(cr, uid, key, inv_type, journal_id, move, context=context)
```

Comment:
@pedrobaeza I see what you mean. Maybe it would be better then to keep the _get_invoice_vals completely as it is in standard Odoo and just pass uid instead of user_id from _invoice_create_line?